src/config/eww_config.rs

```diff
@@ -23,15 +23,41 @@ pub struct EwwConfig {
 }
 
 impl EwwConfig {
+    // TODO: There is definitely a better way to do this with a fold
+    pub fn merge_includes(eww_config: EwwConfig, includes: Vec<EwwConfig>) -> Result<EwwConfig> {
+        let mut eww_config = eww_config.clone();
+        for config in includes {
+            eww_config.widgets.extend(config.widgets);
+            eww_config.windows.extend(config.windows);
+            eww_config.script_vars.extend(config.script_vars);
+            eww_config.initial_variables.extend(config.initial_variables);
+        }
+        Ok(eww_config)
+    }
+
     pub fn read_from_file<P: AsRef<std::path::Path>>(path: P) -> Result<Self> {
-        let content = util::replace_env_var_references(std::fs::read_to_string(path)?);
+        let content = util::replace_env_var_references(std::fs::read_to_string(path.as_ref())?);
         let document = roxmltree::Document::parse(&content)?;
 
-        let result = EwwConfig::from_xml_element(XmlNode::from(document.root_element()).as_element()?.clone());
+        let result = EwwConfig::from_xml_element(XmlNode::from(document.root_element()).as_element()?.clone(), path.as_ref());
         result
     }
 
-    pub fn from_xml_element(xml: XmlElement) -> Result<Self> {
+    pub fn from_xml_element<P: AsRef<std::path::Path>>(xml: XmlElement, path: P) -> Result<Self> {
+        let path = path.as_ref();
+        let includes = match xml.child("includes").context("Error") {
```

Comment:
```suggestion
        let includes = match xml.child("includes") {
```

That's,... unnecessary